src/macaron/slsa_analyzer/analyzer.py

```diff
@@ -128,79 +132,94 @@ def run(self, user_config: dict, sbom_path: str = "", skip_deps: bool = False) -
+                    deps_resolved = self.resolve_dependencies(main_record.context, sbom_path)
+
+                # Merge the automatically resolved dependencies with the manual configuration.
+                deps_config = DependencyAnalyzer.merge_configs(deps_config, deps_resolved)
+
+                # Create a report instance with the record of the main repo.
+                report = Report(main_record)
+
+                duplicated_scm_records: list[Record] = []
+
+                if deps_config:
+                    logger.info("Start analyzing the dependencies.")
+                    for config in deps_config:
+                        dep_status: SCMStatus = config.get_value("available")
+                        if dep_status != SCMStatus.AVAILABLE:
+                            dep_record: Record = Record(
+                                record_id=config.get_value("id"),
+                                description=config.get_value("note"),
+                                pre_config=config,
+                                status=config.get_value("available"),
+                            )
+                            report.add_dep_record(dep_record)
+                            if dep_status == SCMStatus.DUPLICATED_SCM:
+                                duplicated_scm_records.append(dep_record)
+
+                            continue
+                        dep_record = self.run_single(config, analysis, report.record_mapping)
+                        report.add_dep_record(dep_record)
+                else:
+                    logger.info("Found no dependencies to analyze.")
+
+                # Populate the record of duplicated scm dependencies with the
+                # context of analyzed dependencies if available.
+                for dup_record in duplicated_scm_records:
+                    find_ctx = report.find_ctx(dup_record.pre_config.get_value("id"))
+                    dup_record.context = find_ctx
+
+                for record in report.get_records():
+                    if not record.status == SCMStatus.DUPLICATED_SCM:
+                        if record.context:
+                            store_analyze_context_to_db(record.context)
+
+                # Store dependency relations.
+                for parent, child in report.get_dependencies():
+                    parent.component.dependencies.append(child.component)
+
+                # Store the analysis result into report files.
+                self.generate_reports(report)
+
+                # Print the analysis result into the console output.
+                logger.info(str(report))
+
+                db_session.add(analysis)
+                logger.info("Analysis Completed!")
+                return os.EX_OK
+        except sqlalchemy.exc.SQLAlchemyError as error:
+            logger.error("Database error %s", error)
+            logger.critical("The main repository analysis failed. Cannot generate a report for it.")
+            sys.exit(os.EX_DATAERR)
```

Comment:
```suggestion
             return os.EX_DATAERR
```
As the return code of this function will be used in `sys.exit` in this [place](https://github.com/oracle/macaron/blob/9fce105112b0f230d036610d04cc72ed23698a33/src/macaron/__main__.py#L84)

Reply:
This is addressed with recent changes.